docs/book/features/vsphere_csi_migration.md

```diff
@@ -0,0 +1,122 @@
+
+vSphere CSI driver and CNS bring in a lot of features that are not available in the in-tree vSphere volume plugin.
+
+Refer to the following feature comparisons table to know what is added in the vSpehre CSI driver.
+
+| Feature | In-tree vSphere Volume Plugin | vSphere CSI Driver |
+|---------|-------------------------------|--------------------|
+| Block volume (`ReadWriteOnce` access mode) | Supported. Block volumes are backed by vmdks. | Supported. Block volumes are backed by vSphere Improved Virtual Disk(management layer on top of vmdk). |
+| File volume (`ReadWriteMany`/`ReadOnlyMany` access modes) | Not Supported | Supported. File volumes are backed by vSAN file shares. |
+| Dynamic provisioning(with and without SPBM) | Supported for block volumes only | Supported for block and file volumes |
+| Static Provisioning | Supported for block volumes only | Supported for block and file volumes |
+| Expand volume | Not supported | Supported for block volumes |
+| Storage vMotion of block volumes | Not supported | Supported |
+| vSphere UI integration(CNS dashboard, vSAN virtual objects, vSphere space) | Not supported | Supported |
+| [vSphere with Kubernetes Configuration and Management](https://docs.vmware.com/en/VMware-vSphere/7.0/vmware-vsphere-with-kubernetes/GUID-152BE7D2-E227-4DAA-B527-557B564D9718.html) | Not supported | Supported |
+| Independent of vSphere Inventory object names e.g Datacenter name, Datastore name, Storage policy name  | Not Supported | supported |
+| Volume Encryption | Not supported | Supported |
+| Kubernetes Cluster spread across multiple vCenter Servers | Supported | Not supported |
+| Kubernetes Cluster spread across multiple datacenters within a vCenter Server | Supported | Supported |
+| Volume Topology(with `waitForFirstConsumer`) | Supported | Supported |
+| Thick disk provisioning | Supported on all datastore types (vSAN, VVOL, VMFS and NFS) | Supported only on vSAN Datastore using Storage Policy Capability - `Object space reservation` |
+| Raw Block Volume | Supported | Not supported |
+| Inline volumes in Pod spec | Supported | Not supported |
+
+In addition to the above feature comparison, one of the most important things customers need to consider is that Kubernetes has deprecated In-tree vSphere volume plugins and it will be removed in the future Kubernetes releases.
+Kubernetes has provided a seamless procedure to help migrate in-tree vSphere volumes to a vSphere CSI driver. After in-tree vSphere volumes migrated to vSphere CSI driver, All subsequent operations on migrated volumes are performed by the vSphere CSI driver.
+
+## How to migrate to vSphere CSI
+
+In Kubernetes 1.19 release, vSphere CSI Migration is available with `beta` feature-gates.
+
+To try out vSphere CSI migration in beta for vSphere plugin, perform following steps.
+
+1. Upgrade vSphere to 7.0u1.
```

Comment:
We are evaluating extending the support to old vSphere releases as well. Do we need this doc to be check in now? Can this wait?

Reply:
We can wait. but during declaring beta support we already mentioned about this limitation in the release note - 
refer release note on this PR - https://github.com/kubernetes/kubernetes/pull/92816

> Set CSIMigrationvSphere feature gates to beta.
Users should enable CSIMigration + CSIMigrationvSphere features and install the vSphere CSI Driver (https://github.com/kubernetes-sigs/vsphere-csi-driver) to move workload from the in-tree vSphere plugin "kubernetes.io/vsphere-volume" to vSphere CSI Driver.
> Requires: vSphere vCenter/ESXi Version: 7.0u1, HW Version: VM version 15

for the GA release of migration, we can declare the support for prior releases of vSphere.